pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -0,0 +1,631 @@
+	}
+	for _, route := range interfaceSpec.Routes {
+		result.Routes = append(result.Routes, NetworkInterfaceRoute{To: route.To, Via: route.Via, Metric: route.Metric})
+	}
+}
+
+func createNamedNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	finder *find.Finder,
+	interfaceSpec *vmopv1.VirtualMachineNetworkInterfaceSpec) (*NetworkInterfaceResult, error) {
+
+	if interfaceSpec.Network.Kind != "" || interfaceSpec.Network.APIVersion != "" {
+		return nil, fmt.Errorf("network TypeMeta not supported for name network: %v", interfaceSpec.Network.TypeMeta)
+	}
+
+	networkName := interfaceSpec.Network.Name
+	if networkName == "" {
+		return nil, fmt.Errorf("network name is required")
+	}
+
+	backing, err := finder.Network(vmCtx, networkName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find named network %q: %w", networkName, err)
+	}
+
+	return &NetworkInterfaceResult{
+		NetworkID: networkName,
+		Backing:   backing,
+	}, nil
+}
+
+// NetOPCRName returns the name to be used for the NetOP NetworkInterface CR.
+func NetOPCRName(vmName, networkName, interfaceName string, isV1A1 bool) string {
+	var name string
+
+	if isV1A1 {
+		// Old naming convention: each network can really only have 1 NIC.
+		if networkName != "" {
+			name = fmt.Sprintf("%s-%s", networkName, vmName)
+		} else {
+			name = vmName
+		}
+	} else {
+		if networkName != "" {
+			name = fmt.Sprintf("%s-%s-%s", vmName, networkName, interfaceName)
+		} else {
+			name = fmt.Sprintf("%s-%s", vmName, interfaceName)
+		}
+	}
+
+	return name
+}
+
+func createNetOPNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	client ctrlruntime.Client,
+	vimClient *vim25.Client,
+	interfaceSpec *vmopv1.VirtualMachineNetworkInterfaceSpec) (*NetworkInterfaceResult, error) {
+
+	if kind := interfaceSpec.Network.Kind; kind != "" && kind != "Network" {
+		return nil, fmt.Errorf("network kind %q is not supported for VDS", kind)
+	}
+
+	// If empty, NetOP will try to select a namespace default.
+	networkName := interfaceSpec.Network.Name
+
+	netIf := &netopv1alpha1.NetworkInterface{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, false),
+			Namespace: vmCtx.VM.Namespace,
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(vmCtx, client, netIf, func() error {
+		if err := controllerutil.SetOwnerReference(vmCtx.VM, netIf, client.Scheme()); err != nil {
+			// If this fails we likely have an object name collision, and we're in a tough spot.
+			return err
+		}
+
+		netIf.Spec.NetworkName = networkName
+		// NetOP only defines a VMXNet3 type, but it doesn't really matter for our purposes.
+		netIf.Spec.Type = netopv1alpha1.NetworkInterfaceTypeVMXNet3
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	netIf, err = waitForReadyNetworkInterface(vmCtx, client, netIf.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return netIfToResult(vimClient, netIf), nil
+}
+
+func netIfToResult(
```

Comment:
How about netOpNetIfToResult() and ncpNetIfToResult() for the other one below?.

Reply:
Sure